src/diagonal.works/b6/ui/blocks.go

```diff
@@ -463,6 +464,63 @@ func (b *CollectionKeyOrValueBlockJSON) BlockType() string {
 	return b.Type
 }
 
+type HistogramBlockJSON struct {
+	Type  string
+	Range string
+	Value int
+	Total int
+	Index string
+}
+
+func (b HistogramBlockJSON) BlockType() string {
+	return b.Type
+}
+
+func histogramBlocks(c *api.HistogramCollection) (BlocksJSON, error) {
+	keys, values, err := fillKeyValues(c, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var blocks []HistogramBlockJSON
+	total := 0
+	for i, key := range keys {
+		// TODO: Factor out w/ collection-key-or-value fill when we rework blocks
```

Comment:
Leave this as is, because I'll refactor anyway when moving to protos that match the design system. The current factoring was more in service of working out what the design system should be, rather than a sensible outcome.